test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs

```diff
@@ -58,9 +59,8 @@ public static TestServer CreateServer(Action<IApplicationBuilder> builder, strin
         {
             applicationPath = applicationPath ?? WebsitesDirectoryPath;
 
-            // Get current IApplicationEnvironment; likely added by the host.
             var provider = services.BuildServiceProvider();
-            var originalEnvironment = provider.GetRequiredService<IApplicationEnvironment>();
+            var originalEnvironment = PlatformServices.Default.Application;
```

Comment:
Yes, new environment will be picked up, it is executed in WebHostBuilder.BuildHostingServices and default services are added using TryAdd so it won't be overwritten